storm-core/src/jvm/org/apache/storm/utils/ConfigUtils.java

```diff
@@ -0,0 +1,711 @@
+        return (stormRoot + FILE_SEPARATOR + "stormconf.ser");
+    }
+
+    public static String masterInbox(Map conf) throws IOException {
+        String ret = masterLocalDir(conf) + FILE_SEPARATOR + "inbox";
+        try {
+            FileUtils.forceMkdir(new File(ret));
+        } catch (IOException e) {
+            LOG.error("Failed to create dir " + ret, e);
+            throw e;
+        }
+        return ret;
+    }
+
+    public static String masterInimbusDir(Map conf) throws IOException {
+        return (masterLocalDir(conf) + FILE_SEPARATOR + "inimbus");
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorLocalDir mocked = new SetMockedSupervisorLocalDir(dir)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorLocalDir. dir)]
+    //     run test ...)
+    public static class SetMockedSupervisorLocalDir implements Closeable {
+        public SetMockedSupervisorLocalDir(String dir) {
+            mockedSupervisorLocalDir = dir;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorLocalDir = null;
+        }
+    }
+    private static String mockedSupervisorLocalDir = null;
+    public static String supervisorLocalDir(Map conf) throws IOException {
+        if (mockedSupervisorLocalDir != null) {
+            return null;
+        }
+        String ret = absoluteStormLocalDir(conf) + FILE_SEPARATOR + "supervisor";
+        FileUtils.forceMkdir(new File(ret));
+        return ret;
+    }
+
+    public static String supervisorIsupervisorDir(Map conf) throws IOException {
+        return ((supervisorLocalDir(conf) + FILE_SEPARATOR + "isupervisor"));
+    }
+
+    //For testing only
+    // for java
+    // try (SetMockedSupervisorStormDistRoot mocked = new SetMockedSupervisorStormDistRoot(conf)) {
+    //    run test ...
+    // }
+    //
+    // for clojure
+    // (with-open [mock (SetMockedSupervisorStormDistRoot. conf)]
+    //     run test ...)
+    public static class SetMockedSupervisorStormDistRoot implements Closeable {
+        public SetMockedSupervisorStormDistRoot(Map conf) {
+            mockedSupervisorStormDistRoot = conf;
+        }
+        @Override
+        public void close() {
+            mockedSupervisorStormDistRoot = null;
+        }
+    }
+    private static Map mockedSupervisorStormDistRoot = null;
+    public static String supervisorStormDistRoot(Map conf) throws IOException {
+        if (mockedSupervisorStormDistRoot != null) {
+            return null;
+        }
+        return stormDistPath(supervisorLocalDir(conf)); // TODO: no need to forceMake here?, clj does not.
+    }
+
+    public static String supervisorStormDistRoot(Map conf, String stormId) throws IOException {
+        if (mockedSupervisorStormDistRoot != null) {
+            return null;
+        }
+        return supervisorStormDistRoot(conf) + FILE_SEPARATOR + stormId; // TODO: need to (url-encode storm-id)? Not.
```

Comment:
Then it will need to depend on util.clj. Or I will translate the url-encode within this pull request.